test/cpp/api/dataloader.cpp

```diff
@@ -0,0 +1,31 @@
+#include <gtest/gtest.h>
+
+#include <torch/data.h>
+#include <torch/tensor.h>
+
+#include <chrono>
+#include <iostream>
+#include <vector>
+
+using namespace torch::data;
+
+TEST(TestDataLoader, TestA) {
+  auto dataset = datasets::MNIST("test/cpp/api/mnist")
+                     .map(transforms::TensorLambda<>(
+                         [](torch::Tensor input) { return input.mul(10); }))
+                     .map(transforms::Normalize<>(0, 1))
+                     .map(transforms::Stack<>());
+
+  auto data_loader = torch::data::data_loader(
+      std::move(dataset),
+      DataLoaderOptions().batch_size(4).workers(2).enforce_ordering(true));
+
+  for (size_t epoch = 0; epoch < 2; ++epoch) {
+    std::cout << "epoch: " << epoch << std::endl;
+    data_loader->loop([&](Example<> batch) {
```

Comment:
I'd much rather reuse the range-based-for loops instead of coming up with a custom syntax for those things.